### ISSUE TITLE
No APN token received

### ISSUE THREAD

**Issue by ndevr** [raix/push#20]
Hello,

first of all - great project.

I was following these steps: https://github.com/raix/push/wiki/iOS-Micro-Walkthrough but i didn't received an token. I check everything 3x times and created new profiles, certificates etc. but nothing helps. Is there something what i can do to send a token manually from my phone? Is there a special case for the app for deploying to my phone?

Regards ndevr

**Comment by raix** [comment]
make sure to set the `--mobile-server http://x.x.x.x`

**Comment by ndevr** [comment]
Where is you config.push.json located? When I'am locking into the xcode sources at ".meteor/local/cordova-build/platforms/ios/www/application/push.config.web.cordova.js" then the content looks like the default one:

Push.Configure({
	"gcm": {
		"projectNumber": ""
	},
	"production": false,
	"sound": true,
	"badge": true,
	"alert": true,
	"vibrate": true
});

**Comment by raix** [comment]
gcm is missing the apiKey?

**Comment by raix** [comment]
skip the sound/badge/alert/vibrate flags for now...

**Comment by ndevr** [comment]
i did not setup somthing there. Thats what the project was building.

**Comment by raix** [comment]
Moment...

**Comment by ndevr** [comment]
my config.push.json at the root folder of the project:

{
  "apn": {
    "passphrase": "123456",
    "key": "testKey.pem",
    "cert": "testCert.pem",
    "gateway": "gateway.sandbox.push.apple.com"
  },
  "gcm": {
    "apiKey": "",
    "projectNumber": ""
  },
  "production": false,
  "badge": true,
  "sound": true,
  "alert": true,
  "vibrate": true
}

**Comment by raix** [comment]
Create
1. `config.push.json` in your app folder
2. Make sure you have a projectNumber and apiKey for android
3. Make sure to run the app using `--mobile-server`

**Comment by ndevr** [comment]
Can i disable android for now?

**Comment by raix** [comment]
Notes:
* The production flag sets gateway for you
* Dont add sound/badge/alert/vibrate flags in the config for now
* Otherwise it looks ok

Add `Push.debug = true;` on client+server and read the console

**Comment by raix** [comment]
If the server doesnt receive the push token when the app startup the you need to debug the client, if the token is sent to the server then fire up `meteor shell` and look at the data `Push.appCollection.find().fetch();`
If the data is in there and apiKey is set in your config it should work. Begin debugging your network ports could be stopped by firewalls or you might havent set the meteor flag `--mobile-server`

**Comment by raix** [comment]
Also try using `Push.send` in the meteor shell - unless you havent removed the insecure package.

**Comment by ndevr** [comment]
Okay i did everything but nothing happens. Should there be a debug log entry on the xdebug console and could you tell me what is in you ".meteor/local/cordova-build/platforms/ios/www/application/push.config.web.cordova.js" file ? I think the client will not setup correct.

**Comment by raix** [comment]
The https://github.com/raix/push/issues/20#issuecomment-68905933 looks ok - please debug as explained in https://github.com/raix/push/issues/20#issuecomment-68907510 - and read the server and browser console log

You could test if app gets token:
On the client:
```js
Push.addListener(function(token) {
  alert(JSON.stringify(token));
});
```

**Comment by adamgins** [comment]
ps. remember to do --mobile-server http://x.x.x.x:port if you're running this with another port other than 80 ;-)

**Comment by maabed** [comment]
In my case i can only see the token when i run meteor.startup function 
```javascript
Meteor.startup(function () {
 Push.addListener('token', function(token) {
     alert(JSON.stringify(token));
   });
});
```

Using development certification, hope this helps

**Comment by raix** [comment]
The meteor.startup should not matter, it will emit when ready

**Comment by ndevr** [comment]
Nothing helped yet. Strange :/ can you tell me where i can find the config of the apn which is created from config.push.json?

**Comment by ndevr** [comment]
Now I'am receiving messages when i register the handler and doing a manually push-update with the token. Looks like the handler is not registered correctly, but i need abit more time for debugging.

**Comment by dennisharrison** [comment]
I'm just catching up to this issue.  I don't think your client is connected to the server.  Please make sure of that first of all.

I'll be able to pay more attention to this tomorrow.  Don't worry, you're close :)

**Comment by dennisharrison** [comment]
I'm reading this again.

@ndevr Are you saying you have a client token in the collection on your server?

```javascript
Push.appCollection.find().fetch();
```

What is the return?

**Comment by ndevr** [comment]
Yes, i pushed the token into the collection with Meteor.call('raix:push-update') and can send notifications from server/client. But I had to re-register the handlers.

**Comment by raix** [comment]
This should all happen pr. default, the device should register etc. - is this still an issue on the latest version?

**Comment by ClarenceL** [comment]
I just had 1 out of 4 devices not self-register, very odd, but it must still be happening.
